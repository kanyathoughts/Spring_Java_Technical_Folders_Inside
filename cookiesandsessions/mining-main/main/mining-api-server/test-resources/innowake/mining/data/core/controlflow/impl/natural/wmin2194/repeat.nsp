DEFINE DATA
LOCAL
1 #VAR1 (P11.2)
1 EMP VIEW OF EMPLOYEES
 2 PERSONNEL-ID
 2 JOB-TITLE
 2 FIRST-NAME
END-DEFINE

REPEAT
	ADD 1 TO #VAR1
	IF #VAR1 = 1
		ESCAPE BOTTOM
	END-IF
	WRITE NOTITLE '1'
END-REPEAT

REPEAT
	WHILE
		#VAR1
			< 2
	ADD 1 TO #VAR1
	WRITE NOTITLE '2'
END-REPEAT

REPEAT
	ADD 1 TO #VAR1
	WRITE NOTITLE '3'
	WHILE
		#VAR1
			< 3
END-REPEAT

REPEAT
	ADD 1 TO #VAR1
	WRITE NOTITLE '4'
  UNTIL
	#VAR1
		< 4
END-REPEAT

REPEAT
  UNTIL
	#VAR1
		< 5
	ADD 1 TO #VAR1
	WRITE NOTITLE '5'
END-REPEAT


END