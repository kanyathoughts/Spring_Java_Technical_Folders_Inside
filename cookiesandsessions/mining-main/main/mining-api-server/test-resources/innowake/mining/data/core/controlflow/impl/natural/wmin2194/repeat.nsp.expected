 #2 DefineDataStmt
 #12 RepeatStmt, Label: REPEAT, FROM={#26, TestEntryPoint}, TO={#14(LOOP BODY), #30(LOOP END)}
  #14 AddStmt, FROM={#12(LOOP BODY)}, TO={#17}
  #17 IfStmt, Label: IF #VAR1 = 1, FROM={#14}, TO={#23(TRUE), #26(FALSE)}
    #23 EscapeStmt, FROM={#17(TRUE)}, TO={#30}
  #26 WRITE NOTITLE '1', FROM={#17(FALSE)}, TO={#12}
 #30 RepeatStmt, Label: REPEAT WHILE #VAR1 < 2, FROM={#12(LOOP END), #23, #39}, TO={#36(LOOP BODY), #43(LOOP END)}
  #36 AddStmt, FROM={#30(LOOP BODY)}, TO={#39}
  #39 WRITE NOTITLE '2', FROM={#36}, TO={#30}
 #43 RepeatStmt, Label: REPEAT WHILE #VAR1 < 3, FROM={#30(LOOP END), #52}, TO={#49(LOOP BODY), #56(LOOP END)}
  #49 AddStmt, FROM={#43(LOOP BODY)}, TO={#52}
  #52 WRITE NOTITLE '3', FROM={#49}, TO={#43}
 #56 RepeatStmt, Label: REPEAT UNTIL #VAR1 < 4, FROM={#43(LOOP END), #65}, TO={#62(LOOP BODY), #69(LOOP END)}
  #62 AddStmt, FROM={#56(LOOP BODY)}, TO={#65}
  #65 WRITE NOTITLE '4', FROM={#62}, TO={#56}
 #69 RepeatStmt, Label: REPEAT UNTIL #VAR1 < 5, FROM={#56(LOOP END), #78}, TO={#75(LOOP BODY), #82(LOOP END)}
  #75 AddStmt, FROM={#69(LOOP BODY)}, TO={#78}
  #78 WRITE NOTITLE '5', FROM={#75}, TO={#69}
 #82 EndStmt, FROM={#69(LOOP END)}, TO={TestReturnPoint}
